pkg/node/manager/manager.go

```diff
@@ -532,3 +543,41 @@ func (m *Manager) DeleteAllNodes() {
 	m.nodes = map[nodeTypes.Identity]*nodeEntry{}
 	m.mutex.Unlock()
 }
+
+// StartNeighborRefresh spawns a controller which refreshes neighbor table
+// by sending arping periodically. Linux keeps an arp entry in reachable
+// state for (0.5 ~ 1.5) * base_reachable_time_ms, default by 15 to 45 seconds:
+// https://elixir.bootlin.com/linux/v5.7.19/source/net/core/neighbour.c#L113
+// We do the refresh in a similar way.
+func (m *Manager) StartNeighborRefresh(nh datapath.NodeHandler) {
+	var interval time.Duration
+	baseReachableStr, err := sysctl.Read("net.ipv4.neigh.default.base_reachable_time_ms")
+	if err != nil {
+		interval = neighborRefreshBaseInterval
+	} else {
+		baseReachableU32, err := strconv.ParseUint(baseReachableStr, 10, 32)
+		if err != nil {
+			interval = neighborRefreshBaseInterval
+		} else {
+			interval = time.Duration(baseReachableU32) * time.Millisecond
+		}
+	}
+	controller.NewManager().UpdateController("neighbor-table-refresh",
+		controller.ControllerParams{
+			DoFunc: func(context.Context) error {
+				m.mutex.RLock()
+				defer m.mutex.RUnlock()
+				for _, entry := range m.nodes {
+					go func(e *nodeEntry) {
+						n := randGen.Int63n(int64(interval / 2))
+						time.Sleep(interval/2 + time.Duration(n))
+						nh.NodeNeighborRefresh(e.node)
+					}(entry)
```

Comment:
We can avoid spawning N go routines if we simply check if `entry.IsLocal()` first:

```golang
if !entry.IsLocal() {
	go func(e *nodeEntry) {
		n := randGen.Int63n(int64(interval / 2))
		time.Sleep(interval/2 + time.Duration(n))
		nh.NodeNeighborRefresh(e.node)
	}(entry)
}
```